### PR TITLE
Dependencies update

### PR DESCRIPTION
**Pull request by OgreTransporter**
- Upgrade bundled SQLite to 3.46.0
- Upgrade bundled zlib to 1.3.1 (see #4582)
- Upgrade bundled libpng to 1.6.43 (see #4582)
- Upgrade bundled pcre2 to 10.44 (see #4478)

### CODE REVIEW

**Review by aleks-f — CHANGES_REQUESTED** [review]
for some reason, the PR CI is not triggered, but the last commit CI fails massively, please se why

**Comment by OgreTransporter** [comment]
I have already created the fix 6a311fd, but it did not repeat the complete action. Can you trigger this manually?

**Review by aleks-f — CHANGES_REQUESTED** [review]
windows build fails

File: PDF/CMakeLists.txt [hunk 1]
```diff
@@ -94,23 +98,20 @@ POCO_SOURCES(SRCS hpdf
...
+    src/pngread.c
+    src/pngrio.c
+    src/pngrtran.c
+    src/pngrutil.c
+    src/pngset.c
+    src/pngtrans.c
+    src/pngwio.c
+    src/pngwrite.c
+    src/pngwtran.c
+    src/pngwutil.c
```
Comment (matejk) [inline PDF/CMakeLists.txt:114]: Unnecessary change of white space (tabs to spaces).

Is the list of source file shorter by three files intentionally or by accident?

**Comment by aleks-f** [comment]
@OgreTransporter you have some CI failing

**Comment by matejk** [comment]
Replaced with #4676, #4677, #4678

**Pull request closed without merging**